sdk/core/azure-core/src/main/java/com/azure/core/spatial/LineGeometry.java

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.core.spatial;
+
+import java.util.Collection;
+import java.util.Collections;
+
+/**
+ * Represents a geometric line.
+ */
+public final class LineGeometry extends Geometry {
+    private final Collection<GeometryPosition> positions;
+
+    /**
+     * Constructs a geometric line.
+     *
+     * @param positions Geometric positions that define the line.
+     */
+    public LineGeometry(Collection<GeometryPosition> positions) {
+        this(positions, null);
+    }
+
+    /**
+     * Constructs a geometric line.
+     *
+     * @param positions Geometric positions that define the line.
+     * @param properties Additional properties of the geometric line.
+     */
+    public LineGeometry(Collection<GeometryPosition> positions, GeometryProperties properties) {
+        super(properties);
+        this.positions = positions;
+    }
+
+    /**
+     * Unmodifiable representation of the {@link GeometryPosition geometric positions} representing this line.
+     *
+     * @return An unmodifiable representation of the {@link GeometryPosition geometric positions} representing this
+     * line.
+     */
+    public Collection<GeometryPosition> getPositions() {
+        return Collections.unmodifiableCollection(positions);
```

Comment:
Make a defensive copy in the ctor rather than here, otherwise the positions can be changed by the user having a reference to the positions collection that they gave to you.

Reply:
Adding defensive copies of the list in the constructor, also making the store list unmodifiable at the same time.